src/test/java/com/iexec/common/chain/ChainDealTest.java

```diff
@@ -62,26 +63,57 @@
     private static final String FILE2 = "http://test.com/image2.png";
     private static final String FILE3 = "http://test.com/image3.png";
 
+    @Test
+    void testEmptyConstructor() {
+        ChainDeal deal = new ChainDeal();
+        Assertions.assertNull(deal.getBeneficiary());
+        Assertions.assertNull(deal.getBotFirst());
+        Assertions.assertNull(deal.getBotSize());
+        Assertions.assertNull(deal.getCallback());
+        Assertions.assertNull(deal.getChainApp());
+        Assertions.assertNull(deal.getChainCategory());
+        Assertions.assertNull(deal.getChainDataset());
+        Assertions.assertNull(deal.getChainDealId());
+        Assertions.assertNull(deal.getDappOwner());
+        Assertions.assertNull(deal.getDappPrice());
+        Assertions.assertNull(deal.getDataOwner());
+        Assertions.assertNull(deal.getDataPointer());
+        Assertions.assertNull(deal.getDataPrice());
+        Assertions.assertNull(deal.getParams());
+        Assertions.assertNull(deal.getPoolOwner());
+        Assertions.assertNull(deal.getPoolPointer());
+        Assertions.assertNull(deal.getPoolPrice());
+        Assertions.assertNull(deal.getRequester());
+        Assertions.assertNull(deal.getSchedulerRewardRatio());
+        Assertions.assertNull(deal.getStartTime());
+        Assertions.assertNull(deal.getTag());
+        Assertions.assertNull(deal.getTrust());
+        Assertions.assertNull(deal.getWorkerStake());
+    }
+
     @Test
     void shouldReadArgsWithoutJson() {
         DealParams params = stringToDealParams(ARGS);
         assertEquals(ARGS, params.getIexecArgs());
         assertEquals(0, params.getIexecInputFiles().size());
+        assertEquals(0,params.getIexecSecrets().size());
```

Comment:
```suggestion
        assertEquals(0, params.getIexecSecrets().size());
```

Reply:
Fixed in https://github.com/iExecBlockchainComputing/iexec-common/pull/298/commits/86b25b4e02c469cd822bf7e25a2dfd36e10c0e5b